timemachine/fe/system.py

```diff
@@ -86,30 +86,20 @@ def simulate_system(U_fn, x0, num_samples=20000, steps_per_batch=500, num_worker
 
 
 def convert_bps_into_system(bps: Sequence[potentials.BoundPotential]):
-    bond = angle = torsion = nonbonded = chiral_atom = chiral_bond = None
-
-    for bp in bps:
-        if isinstance(bp.potential, potentials.HarmonicBond):
-            bond = bp
-        elif isinstance(bp.potential, potentials.HarmonicAngle):
-            angle = bp
-        elif isinstance(bp.potential, potentials.PeriodicTorsion):
-            torsion = bp
-        elif isinstance(bp.potential, potentials.Nonbonded):
-            nonbonded = bp
-        elif isinstance(bp.potential, potentials.ChiralAtomRestraint):
-            chiral_atom = bp
-        # TODO: uncomment when re-enabling chiral_bond
-        # elif isinstance(bp.potential, potentials.ChiralBondRestraint):
-        #     chiral_bond = bp
-        else:
-            assert 0, "Unknown potential"
-
-    assert bond
-    assert angle
-    assert nonbonded
-
-    return VacuumSystem(bond, angle, torsion, nonbonded, chiral_atom, chiral_bond)
+    assert isinstance(bps[0].potential, potentials.HarmonicBond)
+    assert isinstance(bps[1].potential, potentials.HarmonicAngle)
+    assert isinstance(bps[2].potential, potentials.PeriodicTorsion)  # proper
+    assert isinstance(bps[3].potential, potentials.PeriodicTorsion)  # improper
+    assert isinstance(bps[4].potential, potentials.Nonbonded)  # cursed: NB AllPairs in VacuumSystem
+
+    chiral_atom = ChiralAtomRestraint(np.array([[]], dtype=np.int32).reshape(-1, 4)).bind(
+        np.array([], dtype=np.float64).reshape(-1)
+    )
+    idxs = np.array([[]], dtype=np.int32).reshape(-1, 4)
+    signs = np.array([[]], dtype=np.int32).reshape(-1)
+    chiral_bond = ChiralBondRestraint(idxs, signs).bind(np.array([], dtype=np.float64).reshape(-1))
+
+    return VacuumSystem(bps[0], bps[1], bps[2], bps[3], bps[4], chiral_atom, chiral_bond)
```

Comment:
It seems like this was previously agnostic to the ordering of input potentials, but now requires a specific ordering?

Reply:
Yes but we've only ever used this for things out of the deserializer (host_bps). I tried to work around it using the original code, but the problem is that now we can possibly emit TWO `PeriodicTorsions` for proper and improper, and there's no sane way to determine which is which. The other PR explicitly named them into "proper" and "improper" via a `NamedSystem/VacuumSystem` but that requires a *much* larger refactor.